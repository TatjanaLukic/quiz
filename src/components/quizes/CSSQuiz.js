import React, { useState,useEffect } from "react";
import Result from "../Result";
import QuestionList from "../QuestionList";
import Title from "../Title";
import { cssQuestions } from "../../data/cssQuestions";
import { withRouter} from 'react-router-dom';

const CSSQuiz = () => {
  const [score, setScore] = useState(0);
  const [showResult, setShowResult] = useState(false);
  const [loading, setLoading] = useState(false);

  const setCalculatedScore = calculatedScore => {
    setScore(calculatedScore);
    setShowResult(true);
  };

  return (
    <div style={{ background: "#8f34eb" }}>
      <Title title="CSS Quiz" />
      <div className="ui container">
        {showResult && <Result score={score} />}
        <QuestionList setCalculatedScore={setCalculatedScore} questions={cssQuestions} loading={loading}/>
      </div>
    </div>
  );
};

export default CSSQuiz;
