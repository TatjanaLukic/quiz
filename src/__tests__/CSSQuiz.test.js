import React from 'react';

import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import CSSQuiz from '../components/quizes/CSSQuiz';

configure({adapter: new Adapter()});

describe('<CSSQuiz/>',() => {

it("spanshot matches", () => {
    const tree = shallow(<CSSQuiz />);
    expect(toJson(tree)).toMatchSnapshot();
  });

});