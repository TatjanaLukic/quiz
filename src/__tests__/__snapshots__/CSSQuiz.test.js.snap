// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CSSQuiz/> spanshot matches 1`] = `
<div
  style={
    Object {
      "background": "#8f34eb",
    }
  }
>
  <Title
    title="CSS Quiz"
  />
  <div
    className="ui container"
  >
    <QuestionList
      loading={false}
      questions={
        Array [
          Object {
            "A": "Cascading Style Sheet.",
            "B": "Complete Style Sheet.",
            "answer": "A",
            "category": "basic",
            "id": 1,
            "q": "CSS stands for?",
          },
          Object {
            "A": "Inline, embedded and external style sheet .",
            "B": "Inline and external style sheet.",
            "answer": "A",
            "category": "basic",
            "id": 2,
            "q": "What are the main ways to add CSS to a webpage?",
          },
          Object {
            "A": "Content, padding, border and margin.",
            "B": "Only content, padding and border.",
            "answer": "A",
            "category": "basic",
            "id": 3,
            "q": "Basic elements of box model are:",
          },
          Object {
            "A": "One with higher specificity.",
            "B": "One with lower specificity.",
            "answer": "A",
            "category": "basic",
            "id": 4,
            "q": "When in conflict which design rule has priority?",
          },
          Object {
            "A": "Document flow.",
            "B": "Viewport.",
            "answer": "B",
            "category": "basic",
            "id": 5,
            "q": "Positioning context for element with fixed position is:",
          },
          Object {
            "A": "It is a feature of CSS which is used to style on the given state of an element.",
            "B": "It is a feature of CSS which is used to style the given part of an element.",
            "answer": "A",
            "category": "basic",
            "id": 6,
            "q": "Define a pseudo class in CSS?",
          },
        ]
      }
      setCalculatedScore={[Function]}
    />
  </div>
</div>
`;
